From 002603e46cedd91813e2aa0dd1b2b0e24497079e Mon Sep 17 00:00:00 2001
From: Sebastian Kayser <skayser@opencsw.org>
Date: Sat, 18 Jun 2011 12:54:05 +0200
Subject: [PATCH] On Solaris, Use sys/inttypes.h instead of stdint.h

Required because Solaris < 10 doesn't ship with stdint.h. On Solaris 10,
both (sys/inttypes.h and stdint.h) end up including sys/int_types.h, so
this should also be fine for Solaris 10.
---
 src/Blowfish.c  |    6 ++++++
 src/RIPEMD160.c |    6 ++++++
 src/_counter.h  |    5 +++++
 3 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/src/Blowfish.c b/src/Blowfish.c
index f0d8594..d6e4b7c 100644
--- a/src/Blowfish.c
+++ b/src/Blowfish.c
@@ -27,7 +27,13 @@
  */
 
 #include <assert.h>
+
+#if defined(__sun) || defined(__sun__)
+#include <sys/inttypes.h>
+#else
 #include <stdint.h>
+#endif
+
 #include <string.h>
 #include "Python.h"
 
diff --git a/src/RIPEMD160.c b/src/RIPEMD160.c
index da2e72c..51cfcac 100644
--- a/src/RIPEMD160.c
+++ b/src/RIPEMD160.c
@@ -44,7 +44,13 @@
  */
 
 #include <assert.h>
+
+#if defined(__sun) || defined(__sun__)
+#include <sys/inttypes.h>
+#else
 #include <stdint.h>
+#endif
+
 #include <string.h>
 #include "Python.h"
 
diff --git a/src/_counter.h b/src/_counter.h
index faaf63f..1f2a31c 100644
--- a/src/_counter.h
+++ b/src/_counter.h
@@ -24,7 +24,12 @@
 #ifndef PCT__COUNTER_H
 #define PCT__COUNTER_H
 
+#if defined(__sun) || defined(__sun__)
+#include <sys/inttypes.h>
+#else
 #include <stdint.h>
+#endif
+
 #include "Python.h"
 
 typedef struct {
-- 
1.7.5.4

